app/uk/gov/hmrc/agentaccesscontrol/service/MtdAuthorisationService.scala

```diff
@@ -24,16 +26,37 @@ import uk.gov.hmrc.play.http.HeaderCarrier
 import scala.concurrent.{ExecutionContext, Future}
 
 class MtdAuthorisationService(agenciesConnector: AgenciesConnector,
-                              relationshipsConnector: RelationshipsConnector) {
+                              relationshipsConnector: RelationshipsConnector,
+                              auditService: AuditService) extends LoggingAuthorisationResults {
 
-  def authoriseForSa(agentCode: AgentCode, mtdSaClientId: MtdSaClientId)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Boolean] = {
+  def authoriseForSa(agentCode: AgentCode, mtdSaClientId: MtdSaClientId)
+                    (implicit ec: ExecutionContext, hc: HeaderCarrier, request: Request[_]): Future[Boolean] = {
     agenciesConnector.fetchAgencyRecord(agentCode) flatMap  {
-      case Some(agency) => hasRelationship(agency.arn, mtdSaClientId)
-      case None => Future successful false
+      case Some(agency) => hasRelationship(agency.arn, mtdSaClientId) map { result =>
+        auditDecision(agentCode, mtdSaClientId, result, "arn" -> agency.arn)
```

Comment:
Ah, I'd entirely missed those tests when I was looking for one to steal. I've added some additional unit tests to cover.